sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -103,24 +103,27 @@ class JacksonParser(
    * to a value according to a desired schema. This is a wrapper for the method
    * `makeConverter()` to handle a row wrapped with an array.
    */
-  private def makeRootConverter(dt: DataType): JsonParser => Iterable[InternalRow] = {
+  private def makeRootConverter(dt: DataType): (JsonParser, () => UTF8String)
+    => Iterable[InternalRow] = {
     dt match {
       case st: StructType => makeStructRootConverter(st)
       case mt: MapType => makeMapRootConverter(mt)
       case at: ArrayType => makeArrayRootConverter(at)
     }
   }
 
-  private def makeStructRootConverter(st: StructType): JsonParser => Iterable[InternalRow] = {
+  private def makeStructRootConverter(st: StructType): (JsonParser, () => UTF8String)
+    => Iterable[InternalRow] = {
     val elementConverter = makeConverter(st)
     val fieldConverters = st.map(_.dataType).map(makeConverter).toArray
     val jsonFilters = if (SQLConf.get.jsonFilterPushDown) {
       new JsonFilters(filters, st)
     } else {
       new NoopFilters
     }
-    (parser: JsonParser) => parseJsonToken[Iterable[InternalRow]](parser, st) {
-      case START_OBJECT => convertObject(parser, st, fieldConverters, jsonFilters, isRoot = true)
+    (parser: JsonParser, record: () => UTF8String) =>
```

Comment:
Add `record: () => UTF8String` method, so we can get record in parser. Then throw exception with record.